storm-server/src/main/java/org/apache/storm/scheduler/resource/strategies/scheduling/DefaultResourceAwareStrategy.java

```diff
@@ -18,93 +18,58 @@
 
 package org.apache.storm.scheduler.resource.strategies.scheduling;
 
-import com.google.common.annotations.VisibleForTesting;
-
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Map;
-import java.util.Queue;
-import java.util.Set;
-import java.util.TreeSet;
-
 import org.apache.storm.Config;
-import org.apache.storm.generated.ComponentType;
-import org.apache.storm.scheduler.Cluster;
-import org.apache.storm.scheduler.Component;
-import org.apache.storm.scheduler.ExecutorDetails;
-import org.apache.storm.scheduler.TopologyDetails;
-import org.apache.storm.scheduler.WorkerSlot;
-import org.apache.storm.scheduler.resource.RAS_Node;
-import org.apache.storm.scheduler.resource.RAS_Nodes;
+import org.apache.storm.scheduler.*;
 import org.apache.storm.scheduler.resource.ResourceUtils;
 import org.apache.storm.scheduler.resource.SchedulingResult;
 import org.apache.storm.scheduler.resource.SchedulingStatus;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class DefaultResourceAwareStrategy implements IStrategy {
-    private static final Logger LOG = LoggerFactory.getLogger(DefaultResourceAwareStrategy.class);
-    private Cluster cluster;
-    private Map<String, List<String>> networkTopography;
-    private RAS_Nodes nodes;
+import java.util.*;
 
-    @VisibleForTesting
-    void prepare(Cluster cluster) {
-        this.cluster = cluster;
-        nodes = new RAS_Nodes(cluster);
-        networkTopography = cluster.getNetworkTopography();
-        logClusterInfo();
-    }
+public class DefaultResourceAwareStrategy extends BaseResourceAwareStrategy implements IStrategy {
```

Comment:
Are there any plans to make the Generic one the same as this one?

Reply:
I think that should come in later, ideally there should be a release without GRAS on by default and people can try it out themselves.